src/main/java/mate/academy/dao/BookDaoImplementation.java

```diff
@@ -0,0 +1,105 @@
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO books (title, price) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query,
+                        PreparedStatement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.executeUpdate();
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getLong(1));
+            }
+            return book;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert book into the database: " + book, e);
+        }
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        String query = "SELECT * FROM books WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Book book = extractBook(resultSet);
+                return Optional.of(book);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find book by id: " + id, e);
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Book> findAll() {
+        String query = "SELECT * FROM books";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            ResultSet resultSet = statement.executeQuery();
+            List<Book> books = new ArrayList<>();
+            while (resultSet.next()) {
+                books.add(extractBook(resultSet));
+            }
+            return books;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find all books", e);
+        }
+    }
+
+    @Override
+    public Book update(Book book) {
+        String query = "UPDATE books SET title = ?, price = ? WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection.prepareStatement(query)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            statement.setLong(3, book.getId());
+            if (statement.executeUpdate() > 0) {
+                return book;
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update book: " + book, e);
+        }
+        throw new RuntimeException("No book found to update with id: " + book.getId());
```

Comment:
Instead of throwing a `RuntimeException`, consider using `DataProcessingException` to maintain consistency in error handling across your application. This will help in keeping the error handling uniform and more manageable.